server/index.js

```diff
@@ -51,6 +51,108 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (req.session.cartId) {
+    const sql = `
+      select  "c"."cartItemId",
+              "c"."price",
+              "p"."productId",
+              "p"."image",
+              "p"."name",
+              "p"."shortDescription"
+        from  "cartItems" as "c"
+        join  "products" as "p" using ("productId")
+       where  "c"."cartId" = $1
+  `;
+    const params = [req.session.cartId];
+    db.query(sql, params)
+      .then(result => {
+        return res.json(result.rows);
+      });
+  } else {
+    return [];
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = req.body.productId;
+  if (!productId) {
+    return res.status(400).json({ error: 'productId required' });
+  }
+  if (productId <= 0 || !Number.isInteger(Number(productId))) {
+    return res.status(400).json({ error: 'invalid productId' });
+  }
+  const sql = `
+    select  "price"
+      from  "products"
+     where  "productId" = $1
+  `;
+  const params = [productId];
+  db.query(sql, params)
+    .then(result => {
+      if (result.rows.length < 1) {
+        return next(new ClientError(`cannot find product with productId of ${productId}`, 404));
```

Comment:
Since we're in a promise chain, you need to actually `throw new ClientError(message, statusCode)` instead of calling `next()`. Otherwise, the promise chain will continue to run.